src/engine/CallFixedSize.h

```diff
@@ -91,10 +93,10 @@ auto callLambdaForIntArray(std::array<Int, NumValues> array, auto&& lambda,
 
   // Lambda: call `applyIf` for all the compile-time integers `Is...`. The
   // runtime parameter always is `array`.
-  auto f =
-      [&applyIf]<ArrayType... Is>(ad_utility::ValueSequence<ArrayType, Is...>) {
-        (..., applyIf.template operator()<Is>());
-      };
+  auto f = CPP_template_lambda(&applyIf)(ArrayType... Is)(
+      ad_utility::ValueSequence<ArrayType, Is...>)() {
```

Comment:
To rewrite such things, we probably have to implement a generic 
`apply`  function for the ` ValueSequence` .

Please let me know, if you want to take the challenge, or whether I should prepare the infrastructure for this.